From 39b35f208592fba5e466df4b8f11a61d08046834 Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Thu, 19 Mar 2026 06:36:30 +0000
Subject: [PATCH] exfat: fix double free in delayed_free

jira VULN-163369
cve CVE-2025-38206
commit-author Namjae Jeon <linkinjeon@kernel.org>
commit 1f3d9724e16d62c7d42c67d6613b8512f2887c22

The double free could happen in the following path.

exfat_create_upcase_table()
        exfat_create_upcase_table() : return error
        exfat_free_upcase_table() : free ->vol_utbl
        exfat_load_default_upcase_table : return error
     exfat_kill_sb()
           delayed_free()
                  exfat_free_upcase_table() <--------- double free
This patch set ->vol_util as NULL after freeing it.

	Reported-by: Jianzhou Zhao <xnxc22xnxc22@qq.com>
	Signed-off-by: Namjae Jeon <linkinjeon@kernel.org>
(cherry picked from commit 1f3d9724e16d62c7d42c67d6613b8512f2887c22)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 fs/exfat/nls.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/exfat/nls.c b/fs/exfat/nls.c
index 705710f93e2d..0e05c6b5a286 100644
--- a/fs/exfat/nls.c
+++ b/fs/exfat/nls.c
@@ -804,4 +804,5 @@ int exfat_create_upcase_table(struct super_block *sb)
 void exfat_free_upcase_table(struct exfat_sb_info *sbi)
 {
 	kvfree(sbi->vol_utbl);
+	sbi->vol_utbl = NULL;
 }
-- 
2.47.3

